pkg/operation/botanist/component/vpnseedserver/vpn_seed_server.go

```diff
@@ -716,7 +716,8 @@ func (v *vpnSeedServer) deployService(ctx context.Context, idx *int) error {
 		metav1.SetMetaDataAnnotation(&service.ObjectMeta, "networking.istio.io/exportTo", "*")
 
 		metav1.SetMetaDataAnnotation(&service.ObjectMeta, resourcesv1alpha1.NetworkingPodLabelSelectorNamespaceAlias, v1beta1constants.LabelNetworkPolicyShootNamespaceAlias)
-		utilruntime.Must(gardenerutils.InjectNetworkPolicyNamespaceSelectors(service, metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleIstioIngress}}))
+		utilruntime.Must(gardenerutils.InjectNetworkPolicyNamespaceSelectors(service, metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleIstioIngress}},
+			metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleExposureClassHandler}}))
```

Comment:
```suggestion
			metav1.LabelSelector{MatchExpressions: []metav1.LabelSelectorRequirement{{Key: v1beta1constants.LabelExposureClassHandlerName, Operator: metav1.LabelSelectorOpExists}}}))
```

Unfortunately, the role value is not consistent across all exposure class namespaces, e.g. when they are created for dedicated zones. Please see this commit https://github.com/gardener/gardener/pull/7570/commits/12d5f31f48066f50d836c08d09596a3b73622986#diff-77bddb7bf8b36c35180321a88ccd81424ff675ff29a47d6c372189c95bb27cb9.

Reply:
Adapted in 8823ec9a08de60654eb6535145dec6b1eec41990. I thought if there is a constant defined `GardenRoleExposureClassHandler`, then it should be the same everywhere.

Reply:
That was my assumption in the beginning as well but unfortunately I was told otherwise 😄 However, there is a plan to make this consistent in the future.